### PR TITLE
Store lates auth request and return on demand

### PR DESCRIPTION
**Pull request by Sedemicra**

### CODE REVIEW

File: src/main/java/ee/test_gov/sid/mock/config/SidMockProperties.java [hunk 1]
```diff
@@ -10,10 +10,13 @@
...
         @NotBlank
-        Duration delay, // = Duration.ofSeconds(1);
+        Duration delay,
 
         @NotBlank
-        Duration expiration // = Duration.ofMinutes(5);
+        Duration expiration,
+
+        @NotBlank
+        Boolean storeAuthRequests
```
Comment (karltiirik) [inline src/main/java/ee/test_gov/sid/mock/config/SidMockProperties.java:19]: There needs to be some kind of documentation regarding these properties either here or in README.

File: src/main/java/ee/test_gov/sid/mock/controller/AuthenticationSession.java [hunk 2]
```diff
@@ -27,6 +27,12 @@ public class AuthenticationSession {
     public HttpResponse<Map<String, Object>> authenticationSession(@PathVariable String identifier, @Body AuthenticationInitData body) {
         UUID sessionId = UUID.randomUUID();
         log.info("Received request to authenticate '{}' with session id '{}'", identifier, sessionId);
+
+        try {
+            dataProvider.putRequestData(identifier, body);
```
Comment (karltiirik) [inline src/main/java/ee/test_gov/sid/mock/controller/AuthenticationSession.java:32]: No info level logging?

**Review by karltiirik — CHANGES_REQUESTED** [review]

File: src/main/java/ee/test_gov/sid/mock/config/controller/User.java [hunk 3]
```diff
@@ -30,17 +29,16 @@ public HttpResponse<Map<String, Object>> getUser(@PathVariable String identifier
                 ));
     }
 
-    @Put(consumes = MediaType.APPLICATION_JSON, produces = MediaType.APPLICATION_JSON)
-    public HttpResponse<Map<String, Object>> addUser(@PathVariable String identifier, @Body UserInfo body) {
-
+    @Put(consumes = MediaType.TEXT_PLAIN, produces = MediaType.APPLICATION_JSON)
```
[the remark this answers is not in the document]
Comment (Sedemicra) [reply src/main/java/ee/test_gov/sid/mock/config/controller/User.java:32]: I suppose I failed to test this properly and when actually using discovered it didn't work as expected.

File: src/main/java/ee/test_gov/sid/mock/controller/LatestAuthRequest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import io.micronaut.http.annotation.Produces;
+import javassist.NotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+
+@Slf4j
+@RequiredArgsConstructor
+@Controller("/latest-auth-request")
```
[the remark this answers is not in the document]
Comment (Sedemicra) [reply src/main/java/ee/test_gov/sid/mock/controller/LatestAuthRequest.java:19]: I would like to remove them all from README and replace this part with swagger.

File: src/main/java/ee/test_gov/sid/mock/controller/AuthenticationSession.java (hunk 2 above)
Comment (Sedemicra) [reply src/main/java/ee/test_gov/sid/mock/controller/AuthenticationSession.java:32]: If any logging, then debug or trace level, but which and how much I would decide under logging improvements task.

**Review by karltiirik — APPROVED** [review]

**Pull request merged**